prophyc/tests/generators/test_cpp.py

```diff
@@ -1063,9 +1069,9 @@ def test_generate_swap_declarations():
 namespace prophy
 {
 
-template <> inline C* swap<C>(C* in) { swap(reinterpret_cast<uint32_t*>(in)); return in + 1; }
 template <> A* swap<A>(A*);
 template <> B* swap<B>(B*);
+template <> inline C* swap<C>(C* in) { swap(reinterpret_cast<uint32_t*>(in)); return in + 1; }
```

Comment:
Now the swap declarations are ordered same as models list. Before that we firstly generated swaps or enums and then for structs+unions. I decided to drop that order.